src/androidTest/java/com/owncloud/android/AbstractIT.java

```diff
@@ -43,6 +51,43 @@ public static void beforeAll() {
         client.setUserId(loginName); // for test same as userId
     }
 
+    public String createFile(String name) throws IOException {
+        File tempDir = context.getFilesDir();
+
+        File file = new File(tempDir + File.separator + name);
+        file.createNewFile();
+
+        assertTrue(file.exists());
+
+        return file.getAbsolutePath();
+    }
+
+    /**
+     * Extracts file from AssetManager to cache folder.
+     *
+     * @param fileName Name of the asset file to extract.
+     * @param context  Android context to access assets and file system.
+     * @return File instance of the extracted file.
+     */
+    public static File extractAsset(String fileName, Context context) throws IOException {
+        File extractedFile = new File(context.getCacheDir() + File.separator + fileName);
+        if (!extractedFile.exists()) {
+            InputStream in = null;
+            FileOutputStream out = null;
+            in = context.getAssets().open(fileName);
+            out = new FileOutputStream(extractedFile);
```

Comment:
why do you assign null right before assigning a new reference to in/out variables?

Reply:
Nice founding.
I have no idea what I did there :roll_eyes: